tests/datavolume_test.go

```diff
@@ -0,0 +1,85 @@
+package tests_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	tests "kubevirt.io/kubevirt-ansible/tests/framework"
+	kubev1 "kubevirt.io/kubevirt/pkg/api/v1"
+	ktests "kubevirt.io/kubevirt/tests"
+)
+
+// template parameters
+const rawDataVolumeVMFilePath = "tests/manifests/template/datavolume-vm.yml"
+const rawDataVolumeVMIFilePath = "tests/manifests/template/datavolume-vmi.yml"
+const rawDataVolumeFilePath = "tests/manifests/template/datavolume.yml"
+
+var _ = FDescribe("DataVolume Integration Test", func() {
+	var dataVolumeName, vmName, dstDataVolumeFilePath, url, dstVMIFilePath string
+
+	Context("Datavolume with VM", func() {
+		BeforeEach(func() {
+			dataVolumeName = "datavolume1"
+			vmName = "test-vm-i"
+			dstDataVolumeFilePath = "/tmp/test-datavolume-vm.json"
+			url = "https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img"
+		})
+
+		It("Creating VM and start VMI multiple times will be succeed", func() {
+			tests.ProcessTemplateWithParameters(rawDataVolumeVMFilePath, dstDataVolumeFilePath, "VM_APIVERSION="+kubev1.GroupVersion.String(), "VM_NAME="+vmName, "IMG_URL="+tests.ReplaceImageURL(url), "DATAVOLUME_NAME="+dataVolumeName)
+			tests.CreateResourceWithFilePathTestNamespace(dstDataVolumeFilePath)
+			tests.WaitUntilResourceExists("vm", vmName)
+
+			num := 2
+			By("Starting and stopping the VirtualMachine number of times")
+			for i := 0; i < num; i++ {
+				By(fmt.Sprintf("Doing run: %d", i))
+				tests.StartVirtualMachine(vmName, ktests.NamespaceTestDefault)
+				// Verify console on last iteration to verify the VirtualMachineInstance is still booting properly
+				// after being restarted multiple times
+				if i == num {
+					By("Checking that the VirtualMachineInstance console has expected output")
+					expecter, err := tests.LoggedInCirrosExpecter(vmName, tests.NamespaceTestDefault, 360)
```

Comment:
This will work until the image url is overridden by the CI.  In that case, the console output will not match our expectations.  We either need to stop watching the console and find another way to check the machine is up or we need to disable the ability to override the URL.

Reply:
it is a problem, we can only support some limited numbers of images, but currently `STREAM_IMAGE_URL` is too wide, user can specify any images they want, like Ubuntu, Arch...whatever distribution. so currently we can add a comments said this is limited to be cirros for u/s only temporarily.

it the future, my ideas is we take control limited version of distribution we support. and parse the `STREAM_IMAGE_URL` URL to determine which console to invoke, wdyt?